integration-tests/kafka/src/main/resources/application.properties

```diff
@@ -1,3 +1,13 @@
 quarkus.log.category.kafka.level=WARN
 quarkus.log.category.\"org.apache.kafka\".level=WARN
 quarkus.log.category.\"org.apache.zookeeper\".level=WARN
+
+kafka.bootstrap.servers=localhost:19092
+quarkus.kafka.streams.topics=streams-test-categories,streams-test-customers
+
+# streams options
+quarkus.kafka.streams.application.id=streams-test-pipeline
```

Comment:
I'm getting warnings due to these options being "unknown" in the Quarkus namespace. I'm not quite sure how to handle this, in particular as it's not a fixed set of known options, but it could be any Kafka client/producer/streams option really that'll just be passed through. I thought though that it'd be the "Quarkus way" to have this under some sort of common prefix". Any guidance would be appreciated!

Reply:
Hi @gunnarmorling, here are my2cents about this:

I think you'll need a RUNTIME config 
```
@ConfigRoot(name="kafka", phase=RUN_TIME)
public class KafkaStreamConfig {
    // streams which is a config 
   public KafkaStreams streams;
}

@ConfigGroup
pubic KafkaStreams {
 .. config items here
}

```
See `agroal` extension on how to define them: https://github.com/quarkusio/quarkus/blob/a91a034b855cae3f8d474951cb33e8d05707dc13/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AgroalRuntimeConfig.java#L9. 

Then you can pass around you configs a templates (recorder)  till `KafkaStreamsTopologyManager ` I guess that's where they are used. 

Also you can check hibernate extension which defines several levels of config: https://github.com/quarkusio/quarkus/blob/3e14f37748288c44d531b57d1c80fb7c887b0c02/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java#L13

Reply:
I'm not sure what the proper solution is, but it is possible (although a crappy solution) to remove this false positives: https://github.com/quarkusio/quarkus/blob/00e2bff5b08d9a99743d0104bfd692602bcdeb0e/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L65

Reply:
But looking at the test failure, it looks like something is entirely wrong with the config. I think @dmlloyd needs to give the expert solution here

Reply:
Yes, that's what I'm wondering, too. I'd need to know why the values aren't exposed in native mode. On the "crappy solution" you linked to, the problem is that the set of these options is dynamic, i.e. it could really be any option known to Kafka consumers/producers/streams, and it's not realistic to hard-wire all of them in our code.

Reply:
Sure, but we could just add wildcard support or something to the crappy solution, although that would just make it even crappier probably...

Reply:
Ah, so you already had the discussion here, nice :).

So... about the options, maybe we could at least try to have the most common ones hard wired. It would be more consistent. The issue is that it probably wouldn't be consistent with the more advanced ones. Not sure the notion of "most common ones" make sense here?

For the wildcard solution, you need to use a `Map` in the config object. I think it would work OK but I never used it to point to values, not config objects. So we would have to test it.

What I don't like with all this is that you end up with a bunch of `quarkus` totally inconsistent with our usual naming (with `dashes`) and our convention for types (Duration comes to mind).

Not sure there's an ideal solution there, though.

Reply:
> So... about the options, maybe we could at least try to have the most common ones hard wired. It would be more consistent.

@gsmet, WDYM by hard-wired here?

> you need to use a Map in the config object.

Do you have an example for such config object?

> What I don't like with all this...

I'm open to alternatives. I moved this under the "io.quarkus" namespace because it felt like the way to do it, but we also could just support giving the options with their original names in application.properties. Is there precedent for that?

I'm not sure about the dashes and types part; keep in mind all these are options known to and used by Kafka Streams users. If we alter names or value representations, it'll create friction when migrating existing code or simply applying the name/value someone has read in the upstream docs or other material.